test/unit/query/builder.js

```diff
@@ -7854,6 +7854,29 @@ describe('QueryBuilder', () => {
     );
   });
 
+  it('row_number with object', function () {
```

Comment:
can you also add an integration test?

Reply:
I can try to look into it, haven't really written integration tests before, probably will just copy one of the other ones and get it to fit the function

Reply:
@kibertoad after looking through the `test/integration` folder, I can't find anything previously for `rowNumber`, `denseRank`, `rank` in any of the files, so I'm unsure of how I should do it in that case.

Reply:
You are right, we only have unit tests for analytic functions now